frontend/src/layouts/AppLayout/AppLayout.tsx

```diff
@@ -909,20 +930,73 @@ export const AppLayout = ({ children }: LayoutProps) => {
-                  name="name"
-                  defaultValue=""
-                  render={({ field, fieldState: { error } }) => (
-                    <FormControl
-                      label="Project Name"
-                      isError={Boolean(error)}
-                      errorText={error?.message}
-                    >
-                      <Input {...field} placeholder="Type your project name" />
-                    </FormControl>
-                  )}
-                />
+                <div className="flex gap-2">
+                  <Controller
+                    control={control}
+                    name="name"
+                    defaultValue=""
+                    render={({ field, fieldState: { error } }) => (
+                      <FormControl
+                        label="Project Name"
+                        isError={Boolean(error)}
+                        errorText={error?.message}
+                        className="flex-1"
+                      >
+                        <Input {...field} placeholder="Type your project name" />
+                      </FormControl>
+                    )}
+                  />
+                  <Controller
+                    control={control}
+                    name="templateId"
+                    render={({ field: { value, onChange } }) => (
+                      <OrgPermissionCan
+                        I={OrgPermissionActions.Read}
+                        a={OrgPermissionSubjects.ProjectTemplates}
+                      >
+                        {(isAllowed) => (
+                          <FormControl
+                            label="Project Template"
+                            icon={<FontAwesomeIcon icon={faInfoCircle} size="sm" />}
+                            tooltipText={
+                              <>
+                                <p>
+                                  Create this project from a template to provision it with custom
+                                  environments and roles.
+                                </p>
+                                {subscription && !subscription.projectTemplates && (
+                                  <p className="pt-2">Project templates are a paid feature.</p>
+                                )}
+                              </>
+                            }
+                          >
+                            <Select
+                              defaultValue={DefaultProjectTemplateIdentifier}
+                              placeholder={DefaultProjectTemplateIdentifier}
+                              isDisabled={!isAllowed || !subscription?.projectTemplates}
+                              value={value}
+                              onValueChange={onChange}
+                              className="w-44"
+                            >
+                              <SelectItem value={DefaultProjectTemplateIdentifier}>
+                                {DefaultProjectTemplateIdentifier}
```

Comment:
This feels a bit weird pattern. May be we can add a flag in api something like default ones/reserved . Because i feel there could be more future templates also. Like one for monorepo setup etc

Reply:
yeah good idea, I'll make `templatedId` uuid or enum so we can expand infisical templates in the future ie default if `undefined` or `"default"` otherwise query for template by ID